src/nvim/CMakeLists.txt

```diff
@@ -221,15 +221,33 @@ function(get_preproc_output varname iname)
   endif()
 endfunction()
 
-set(NVIM_VERSION_GIT_H ${PROJECT_BINARY_DIR}/cmake.config/auto/versiondef_git.h)
+execute_process(
+  COMMAND git rev-parse --is-inside-work-tree
```

Comment:
>  Keep things simple and obvious.

I agree this may not be too obvious, but we still need to cover the two separate cases either way

> 2\. why can't this be done in `GenerateVersion.cmake`? Just generate an empty file or something.

I wanted to avoid re-generating `versiondef_git.h` on every build for those specific workflows, but maybe that's not a big deal.

Reply:
tarball or non-git repo is not the common case so it doesn't matter. What matters is fixing https://github.com/neovim/neovim/pull/19289#discussion_r925717296 and keeping the code simple and clear.